### PR TITLE
Proper usage of update_every in LdaMulticore

### PR DESCRIPTION
**Pull request by jonaschn**
LdaMulticore offers a different API compared to `LdaModel`.
This is unnecessary complex.
Neither the removed `batch` nor the `update_every` parameters are documented for `LdaMulticore`.
However, LdaMulticore inherits `update_every` from `LdaMulticore` anyway.

This PR removes the `batch` parameter and uses the `update_every` parameter analog to it's usage in `LdaModel`:
`batch=True` becomes `update_every=0`
`batch=False` becomes `update_every=1`
New: update_every >= 2 (as used in ldamodel.py)

### CODE REVIEW

**Comment by piskvorky** [comment]
You're on a roll @jonaschn :)

File: gensim/models/atmodel.py [hunk 1]
```diff
@@ -23,6 +23,9 @@
 <https://arxiv.org/abs/1207.4169>`_. The model correlates the authorship information with the topics to give a better
 insight on the subject knowledge of an author.
 
+.. _'Online Learning for LDA' by Hoffman et al.: online-lda_
```
Comment (mpenkov) [inline gensim/models/atmodel.py:26]: I think these changes are not related to this PR. Can you revert them? Alternatively, you may rebase off upstream/develop, these changes have already been merged in there.

File: gensim/models/ldamulticore.py [hunk 2]
```diff
@@ -229,14 +230,20 @@ def update(self, corpus, chunks_as_numpy=False):
 
         self.state.numdocs += lencorpus
 
-        if self.batch:
+        # Same as in LdaModel but self.workers (processes) is used instead of self.numworkers (machines)
+        if self.update_every:
+            updatetype = "online"
```
Comment (mpenkov) [inline gensim/models/ldamulticore.py:235]:
```suggestion
            updatetype = "online (single-pass)" if self.passes == 1 else "online (multi-pass)"
```

File: gensim/models/atmodel.py (hunk 1 above)
Comment (jonaschn) [reply gensim/models/atmodel.py:26]: I messed up with Github's new fetch upstream feature and opened a new PR.

File: gensim/models/ldamulticore.py (hunk 2 above)
Comment (jonaschn) [reply gensim/models/ldamulticore.py:235]: I addressed your feedback in #3140
